### DISCUSSION TITLE
How to change the framerate

### DISCUSSION THREAD

**Discussion by xiaoyaodingchunqiu** [awslabs/amazon-kinesis-video-streams-producer-sdk-cpp#973] [Q&A] [answered]
hi!
    we want do some sampling and need change the framerate to save  some traffic.
    How to change the framerate in the code when run kvs_gstreamer_sample 
   where it is?
thanks!

**Comment by hassanctech** [comment, marked as answer]
KVS does not do any transcoding so you cannot change the frame rate directly.  You can always use the `videorate` and `audiorate` GStreamer elements to do this though:

https://gstreamer.freedesktop.org/documentation/videorate/index.html?gi-language=c
https://gstreamer.freedesktop.org/documentation/audiorate/index.html?gi-language=c

Frame rate for example in a codec like `h264` is embedded in the stream itself in the sps/pps, so you will need to use one of the elements above to make sure the frame it is correct *before* it reaches `kvssink` once it reaches the sink element you cannot just tell the sink element to change it, it must be correctly set before it hits the sink, and using the elements above will accomplish that.
